### ISSUE TITLE
Permissions broken for files?

### ISSUE THREAD

**Issue by CWalkden** [tendenci/tendenci#1245]
Using the Files module, we have some files set up with permissions so that only administrators can access them.  For instance:
![image](https://github.com/tendenci/tendenci/assets/64681720/efee7e1f-4eb7-447e-8191-bd7d7924ef72)

This file is now accessible to non-administrators!
A person without an account can't access it, but a member can.  If they paste the URL for the file into their browser then the file downloads.

I've tested this on 14.8 and 14.8.1

**Comment by jennyq** [comment]
Are you talking about the file url like this /files/123/? I've just tested it, it works fine for me.

**Comment by CWalkden** [comment]
> Are you talking about the file url like this /files/123/? I've just tested it, it works fine for me.

Yes, that's correct.
I just tried it again from scratch.  Here's what I did:

1. Created a text file on my computer with a single character in it (Tendenci doesn't allow empty files).
2. As an Administrator, went to the Files module (/files/).
3. In the menu, chose Files -> Add a File
4. Chose the text file on my computer.  
5. Cleared the checkbox "Public can view".  All other permissions checkboxes are cleared.
6. Saved (uploaded) the file.
7. Copied the file's URL into the clipboard: https://test.aeva.asn.au/files/2444/
8. Opened a new private window in my browser.
9. Pasted in the URL - Tendenci asks to log in.
10. Logged in as a non-administrator user.  This user is Staff, and is in groups with various privileges including privileges for the Files module.
11. The file opened in the browser.  It should have said that I don't have permission.

**Comment by CWalkden** [comment]
I just tried this again, logging in as a user who does not have a membership and is not staff.  This user received the appropriate message:
![image](https://github.com/tendenci/tendenci/assets/64681720/77a035c1-6ee6-4839-984f-8b71c2e23eca)

**Comment by CWalkden** [comment]
I tried again as a member, and the member also can't access the file.
I then changed their security level to Staff, and they were able to access the file.  So I think it is related to the security level.

**Comment by jennyq** [comment]
The reason the user in your Staff group is able to access the file is because the Staff group has the privileges for the Files module. If you remove the permissions from the Staff group, they shouldn't be able to access the file.

**Comment by CWalkden** [comment]
> The reason the user in your Staff group is able to access the file is because the Staff group has the privileges for the Files module. If you remove the permissions from the Staff group, they shouldn't be able to access the file.

How do we adjust the permissions for the Staff group?  There is no group with this name in /groups/

To be clear, I'm talking about the Security Level setting, found in the Profile:
![image](https://github.com/tendenci/tendenci/assets/64681720/8f839ea4-ef85-48e7-9487-55690930e67c)
The options are: User, Staff, Superuser.

The only reason we have people with the Staff permission is that we want them to be able to send Newsletters.  We've found that they need to be Staff to do this.  We want to be able to restrict these people from viewing certain files, but we still want them to be able to use the Files module.

**Comment by jennyq** [comment]
You mean Staff users have the files permissions automatically? That can **not** be. Please check what groups this user is in. There must be a group with the files permissions.

**Comment by CWalkden** [comment]
> You mean Staff users have the files permissions automatically? That can **not** be. Please check what groups this user is in, and check the permissions for those groups. Also, check what permissions are assigned to this profile.

You are correct!
I repeated this test today, and got different results.  Perhaps I got my browser windows mixed up previously.  Above I stated that changing to Staff level gave access to the file, but _this is not correct_.

As suggested, the members who have access to the file are members of a group with the following permissions:
![image](https://github.com/tendenci/tendenci/assets/64681720/1c2740a3-cd5a-4ac2-8a05-ceeb2e2e10bf)

I removed the permission "Can view file", and this removed their access to the file.
Why was this permission selected?  These people need access to the Files module, to upload files occasionally.  This permission looks like it would give them access to the Files module, so I reckon that is why it was applied to the group.  But I removed the permission and the user could access the Files module correctly, they just couldn't see those restricted files.
Perhaps this permission would be better named "Can view all files"?
